src/components/Modal/ModalConnectNetworks.tsx

```diff
@@ -52,13 +55,27 @@ const ModalConnectNetworks = () => {
     account: string,
     setErrorAccount: (value: string) => void
   ) => {
-    const chainName = convertPayload.selectedTokenOut?.blockchain
-    if (chainName === "eth") {
-      if (!ethers.utils.isAddress(account)) {
-        setErrorAccount("Invalid wallet address.")
-      } else {
-        setErrorAccount("")
-      }
+    const result = parseDefuseAsset(
+      convertPayload.selectedTokenOut?.defuse_asset_id ?? ""
+    )
+    const blockchain = result?.blockchain ?? ""
+    switch (blockchain) {
+      case BlockchainEnum.Eth:
+        if (!ethers.utils.isAddress(account)) {
+          setErrorAccount("Invalid wallet address.")
+        } else {
+          setErrorAccount("")
```

Comment:
We can reset error on any rechek, this avoid duplication in case there is no error

Reply:
Not exactly as moving this additional `setErrorAccount` out of scope of switch will cause batching of two state: one from out of switch and another from switch case.